libevmasm/EVMAssemblyStack.cpp

```diff
@@ -36,7 +36,7 @@ void EVMAssemblyStack::parseAndAnalyze(std::string const& _sourceName, std::stri
 	solAssert(!m_evmAssembly);
 	m_name = _sourceName;
 	Json::Value assemblyJson;
-	solRequire(jsonParseStrict(_source, assemblyJson), AssemblyImportException, "Could not parse JSON file.");
+	solRequire(jsonParse(_source, assemblyJson), AssemblyImportException, "Could not parse JSON file.");
```

Comment:
How non-strict is `jsonParse()`? If the difference is just the `null`, it's fine, but I have a feeling that this will affect also some other aspects of parsing that we'd rather keep strict :)

Reply:
Not restrictive at all, it uses the default settings. Please see [setDefaults()](https://open-source-parsers.github.io/jsoncpp-docs/doxygen/class_json_1_1_char_reader_builder.html#a03ff031e06aabff989ab4addc87294ab) and [strictMode()](https://open-source-parsers.github.io/jsoncpp-docs/doxygen/class_json_1_1_char_reader_builder.html#a9c19e3c5475f9072d527810d4bf56749) for a comparison.

But we could change the `strictRoot` field in the strict settings to `false`, which will allow us to accept `Json::null`, however we need to double check possible consequences of this. The commit https://github.com/ethereum/solidity/commit/772ff3e8cc191aeed6b6517db300ccbd0b52a23e was just an example of the issue.

Reply:
I reverted the changes anyway ;)

Reply:
```diff
-  (*settings)["collectComments"] = true;
-  (*settings)["allowComments"] = true;
-  (*settings)["allowTrailingCommas"] = true;
-  (*settings)["strictRoot"] = false;
+  (*settings)["allowComments"] = false;
+  (*settings)["allowTrailingCommas"] = false;
+  (*settings)["strictRoot"] = true;
   (*settings)["allowDroppedNullPlaceholders"] = false;
   (*settings)["allowNumericKeys"] = false;
   (*settings)["allowSingleQuotes"] = false;
   (*settings)["stackLimit"] = 1000;
-  (*settings)["failIfExtra"] = false;
-  (*settings)["rejectDupKeys"] = false;
+  (*settings)["failIfExtra"] = true;
+  (*settings)["rejectDupKeys"] = true;
   (*settings)["allowSpecialFloats"] = false;
   (*settings)["skipBom"] = true;
```

I see that there are some important differences. For example `rejectDupKeys` does not sound like something we'd like to disable. And even things like comments or trailing commas would be something we could not roll back later in a non-breaking way when we finally switch to a better maintained JSON library.

Using `strictMode()` with just the `strictRoot` flag flipped would probably be acceptable, but still, this is really something better done in a separate PR, because here there are too many other concerns mixed in. We do need to consider whether the new JSON parser will allow that or not.